Consolidate Window Configuration

combatwombat16 opened the pull request:
Summary
This PR consolidates window management into a Single Source of Truth and significantly simplifies the overlay interaction logic. We have removed the complex custom cursor monitoring system in favor of standard Tauri window behaviors, ensuring better performance and stability across platforms.

Key Changes
1. Unified Window Configuration
- Single Source of Truth: Introduced shared/window-config.json to define all window labels and titles.
- Implementation: Both Frontend (window-titles.ts) and Backend (config.rs) now consume this shared config, eliminating hardcoded strings and potential mismatches.
2. Overlay & Interaction Overhaul
- Removed Custom Polling: Deleted the high-frequency (20Hz) cursor monitoring threads in linux.rs/windows.rs and the useClickThrough frontend hook.
- Standardized Behavior: Restored standard set_ignore_cursor_events(true) for the Main window. Overlays now rely on native window transparency rather than complex backend coordinate tracking.
- Linux Fixes: Implemented a compositor "resize kick" to eliminate visual ghosting for transparent windows.
3. Cleanup & Optimization
- Performance: Reduced CPU overhead by removing background polling threads.
- Sanitization: Removed unused Tauri commands (start_click_through_poll), capabilities, dependencies, and verbose debug logs.

Verification
- Config: Verify window titles match window-config.json.
- Interaction: Confirm the main overlay is click-through to the game, while specific UI windows (Chat, Settings) remain interactive.
- Stability: Verify no visual artifacts on Linux and clean build/startup with no console errors.

combatwombat16 committed:
feat(backend): unify backend cursor monitoring & remove frontend polling
- Refactor useClickThrough to use Mutation/Resize Observers instead of polling.
- Update ChatButton and ChatOverlayWidget to use unified backend monitoring.
- Remove legacy useChatClickThrough and useWindowClickThrough hooks.
- Implement Force Focus and start/stop logic in Rust backend.

combatwombat16 committed:
refactor: standardize window titles and cleanup debug logs

- Implemented [src/common/window-config.json] as a single source of truth for labels and titles.
- Updated [src/lib/window-titles.ts] to consume the shared JSON configuration.
- Refactored Rust backend modules ([lib.rs], [commands.rs]) to load [window-config.json] at runtime.
- Removed legacy hardcoded "PD2 Trader" strings and redundant debug logs.
- Updated `default.json` capabilities to include window title permissions.

combatwombat16 committed:
chore(merge): merge in main

combatwombat16 committed:
refactor: remove custom cursor monitoring and restore default click-through

- Removed `useClickThrough` hook and all associated frontend registration logic.
- Removed complex backend cursor polling threads and logic from [linux.rs] and [windows.rs].
- Removed exposed monitoring commands from [commands.rs] and [lib.rs].

combatwombat16 committed:
chore(cleanup): remove defunct imports, references and methods

errolgr commented on src/components/custom/ChatButton.tsx:
```diff
@@ -62,58 +62,31 @@ export const ChatButton: React.FC<ChatButtonProps> = ({
-    };
+    // We can also listen to the standard window blur event as a safety net
+    window.addEventListener('blur', handleMouseLeaveWindow);
+    return () => window.removeEventListener('blur', handleMouseLeaveWindow);
```
We should remove these changes as well since we are removing the click through legacy stuff.

combatwombat16 committed:
refactor(landing): consolidate window management and extract hooks
- decompose LandingPage.tsx into specialized hooks (useItemActions, useChatManager, etc)
- extract generic window helpers to src/lib/window-helpers.ts for global reuse
- create useDiabloFocus hook to centralize focus detection
- refactor ItemOverlayWidget to use event-driven window management
- centralize window references to prevent synchronization issues

combatwombat16 committed:
revert(landing-page): revert erroneous commit

combatwombat16 committed:
chore(merge): catchup to main

combatwombat16 committed:
chore(cleanup): remove hanging bits of code from cursor tracking impl

combatwombat16 committed:
chore(cleanup): remove hanging bits of code from cursor tracking impl
